pytorch_lightning/metrics/classification/stat_scores.py

```diff
@@ -0,0 +1,250 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Any, Callable
+
+import torch
+from pytorch_lightning.metrics import Metric
+from pytorch_lightning.metrics.functional.stat_scores import _stat_scores_update, _stat_scores_compute
+
+
+class StatScores(Metric):
+    """Computes the number of true positives, false positives, true negatives, false negatives.
+    Related to `Type I and Type II errors <https://en.wikipedia.org/wiki/Type_I_and_type_II_errors>`__
+    and the `confusion matrix <https://en.wikipedia.org/wiki/Confusion_matrix#Table_of_confusion>`__.
+
+    The reduction method (how the statistics are aggregated) is controlled by the
+    ``reduce`` parameter, and additionally by the ``mdmc_reduce`` parameter in the
+    multi-dimensional multi-class case.
+
+    Accepts all inputs listed in :ref:`metrics:Input types`.
+
+    Args:
+        threshold:
+            Threshold probability value for transforming probability predictions to binary
+            (0,1) predictions, in the case of binary or multi-label inputs. If not set it
+            defaults to 0.5.
+
+        top_k:
+            Number of highest probability entries for each sample to convert to 1s - relevant
+            only for inputs with probability predictions. If this parameter is set for multi-label
+            inputs, it will take precedence over threshold. For (multi-dim) multi-class inputs,
+            this parameter defaults to 1.
+
+            Should be left unset (``None``) for inputs with label predictions.
+
+        reduce:
+            Defines the reduction that is applied. Should be one of the following:
+
+            - ``'micro'`` [default]: Counts the statistics by summing over all [sample, class]
+              combinations (globally). Each statistic is represented by a single integer.
+            - ``'macro'``: Counts the statistics for each class separately (over all samples).
+              Each statistic is represented by a ``(C,)`` tensor. Requires ``num_classes``
+              to be set.
+            - ``'samples'``: Counts the statistics for each sample separately (over all classes).
+              Each statistic is represented by a ``(N, )`` 1d tensor.
+
+            Note that what is considered a sample in the multi-dimensional multi-class case
+            depends on the value of ``mdmc_reduce``.
+
+        num_classes:
+            Number of classes. Necessary for (multi-dimensional) multi-class or multi-label data.
+
+        ignore_index:
+            Specify a class (label) to ignore. If given, this class index does not contribute
+            to the returned score, regardless of reduction method. If an index is ignored, and
+            ``reduce='macro'``, the class statistics for the ignored class will all be returned
```

Comment:
why some code in the docs is ` and other `` and some ranges without anything?

Reply:
I'll check for consistency. If it's a range I am fine that it is without anything (or should we put ` ? ), as it is not code.

Reply:
I went over it now - now double backticks should be consistently used in docstrings for either python primitives (None, False, strings), dimension names (N, C, ...) or parameter/method names. Single backticks are used in error messages - I think that's fine, it's just to clearly let the user know what is the parameter in question. For ranges, I only found "(0,1)", which was not actually a range - I changed it to "(0 or 1)" for clarity. Let me know if there's anything else that needs to be changed